src/AzureAuth/CommandMain.cs

```diff
@@ -262,10 +262,45 @@ private int GetToken()
             try
             {
                 ITokenFetcher tokenFetcher = this.TokenFetcher();
-                TokenResult tokenResult = (this.tokenFetcherOptions.Scopes == null
-                    ? tokenFetcher.GetAccessTokenAsync(this.CombinedAuthMode)
-                    : tokenFetcher.GetAccessTokenAsync(this.tokenFetcherOptions.Scopes, this.CombinedAuthMode))
-                    .Result;
+                TokenResult tokenResult = null;
+
+                // When running multiple AzureAuth processes with the same tenant and resource ID,
+                // They may prompt many times, which is annoying and unexpected.
+                // Use Mutex to ensure that only one process can access the corresponding resource at the same time.
+                string lockName = "Global\\"
+                    + this.Resource
+                    + this.Client
+                    + this.Tenant;
+
+                // First parameter InitiallyOwned indicated whether this lock is owned by current thread.
+                // It should be false otherwise a dead lock could occur.
+                using (Mutex mutex = new Mutex(false, lockName))
+                {
+                    try
+                    {
+                        // Wait for other session exit.
+                        mutex.WaitOne();
+                    }
+
+                    // AbandonedMutexException could be thrown if another process exit without releasing mutex correctly.

```

Comment:
There's a small grammatical correction that can be made here.
```suggestion
                    // An AbandonedMutexException could be thrown if another process exits without releasing the mutex correctly.

```